### PR TITLE
shared/net: improve direct use of netip.Addr

### PR DESCRIPTION
**Pull request by amery**

### CODE REVIEW

File: shared/net/addrs.go [hunk 1]
```diff
@@ -141,23 +157,21 @@ func GetIPAddresses(ifaces ...string) ([]netip.Addr, error) {
...
-			}
+	return out, nil
+}
 
-			if ip, ok := netip.AddrFromSlice(s); ok {
-				out = append(out, ip)
-			}
+func appendNetIPAsIP(out []netip.Addr, addrs ...net.Addr) []netip.Addr {
+	for _, netAddr := range addrs {
+		addr, err := netip.ParseAddr(netAddr.String())
```
Comment (karasz) [inline shared/net/addrs.go:168]: this line has a "bug" on Linux. On Linux, as opposed to Windows, `net.InterfaceAddrs()` will return addresses with CIDR notation. netip.ParseAddr is chocking on the CIDR notation and returns invalid address.
the fix could be:
```go
func appendNetIPAsIP(out []netip.Addr, addrs ...net.Addr) []netip.Addr {
	for _, netAddr := range addrs {
		netAddrStr := strings.Split(netAddr.String(), "/")[0]
		addr, err := netip.ParseAddr(netAddrStr)
		if err == nil && addr.IsValid() {
			out = append(out, addr)
		}
	}

	return out
}
```

**Review by karasz — CHANGES_REQUESTED** [review]
Please see code comments

File: shared/net/addrs.go (hunk 1 above)
Comment (karasz) [reply shared/net/addrs.go:168]: Or even an one-liner:
```go
addr, err := netip.ParseAddr(strings.Split(netAddr.String(), "/")[0])
```

**Comment by amery** [comment]
when back to the AddrFromSlice() approach as getting the IP from a net.IPNet is cleaner than splitting the strings before passing them to ParseAddr().

GetNetIPAddresses() fixed to Unmap() 4in6 addresses first

**Comment by karasz** [comment]
Unmap() can be applied to all Addrs, in the case of an IPv4 it is a no-op, I guess cheaper than an `if ` . Internally Unmap() checks for `Is4In6`

**Review by karasz — APPROVED** [review]
LGTM

**Pull request merged**